keepassxc-browser/content/username-field.js

```diff
@@ -54,7 +54,16 @@ const createIcon = function(target, databaseClosed) {
 
     const field = target;
     const className = getIconClassName(databaseClosed);
-    const size = (field.offsetHeight > 28) ? 24 : 16;
+
+    // Size the icon dynamically, but not greater than 24 or smaller than 14
+    const size = (field.offsetHeight > 28) ? 24 : Math.max((field.offsetHeight - 4), 14);
```

Comment:
This could be simplified to: `Math.max(Math.min(24, field.offsetHeight - 4), 14)`